src/main/daml/Daml/Finance/Interface/Util/Disclosure.daml

```diff
@@ -79,27 +79,27 @@ interface Disclosure where
     -- Any party can undisclose itself. None is returned if no update is needed.
     with
       disclosers : Parties
-        -- ^ Party calling this choice.
+        -- ^ Parties calling this choice.
       observersToRemove : (Text, Parties)
         -- ^ Observer context to remove.
     controller disclosers
     do
       assertMsg "controller must be authorized to RemoveObservers when removing other parties" $
-        disclosers == snd observersToRemove ||
+        not (S.null $ S.intersection disclosers (snd observersToRemove)) ||
```

Comment:
```suggestion
        not (S.null $ disclosers `S.intersection` snd observersToRemove) ||
```
I dont think you need the brackets around `(snd observersToRemove)` as per line 95.